docker/entrypoint.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+    local var="$1"
+    local fileVar="${var}_FILE"
+    local def="${2:-}"
+    if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+        echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+        exit 1
+    fi
+    local val="$def"
+    if [ "${!var:-}" ]; then
+        val="${!var}"
+    elif [ "${!fileVar:-}" ]; then
+        val="$(< "${!fileVar}")"
+    fi
+    export "$var"="$val"
+    unset "$fileVar"
+}
+
+file_env 'IOTA_AUTH_USER'
+file_env 'IOTA_AUTH_PASSWORD'
+file_env 'IOTA_AUTH_ENABLED'
+
+if [[ $IOTA_AUTH_ENABLED ==  "false" ]]; then
+ echo "***********************************************"
+ echo "WARNING: It is recommended to enable authentication for secure connection"
+ echo "***********************************************"
+else
+    if [[ $IOTA_AUTH_USER ==  "" ]] || [[ $IOTA_AUTH_PASSWORD == "" ]]; then
+        echo "***********************************************"
+        echo "WARNING: It is recommended to set IOTA Auth credentials when using authentication"
```

Comment:
The Warning should read:

> WARNING: Default IoT Agent Auth credentials have not been overridden

And nothing more - we can't infer too much from this.
The connection to the Context broker will (probably) **Fail** directly after this.

Docker is flexible so I know there is a workaround where you could supply a `VOLUME`, but its more likely that you've set Auth on and not supplied the Username/Password

Reply:
Fixed with https://github.com/telefonicaid/iotagent-json/pull/380/commits/c33b866240d67dcd3071bf277b278e1e617c5c9f